File: Frontend/src/services/adminProfileService.js
import axiosInstance from "../api/axiosInstance";

export const adminProfileService = {

    getProfile: async () => {

        const response =
            await axiosInstance.get("/admin/profile");

        return response.data;
    },

    updateProfile: async (profileData) => {

        const response =
            await axiosInstance.put(
                "/admin/profile",
                profileData
            );

        return response.data;
    }

};

File: Backend/src/main/java/com/project/autoserve/service/AdminProfileService.java
package com.project.autoserve.service;

import com.project.autoserve.dto.profile.AdminProfileResponseDTO;
import com.project.autoserve.dto.profile.UpdateProfileRequestDTO;

public interface AdminProfileService {

    AdminProfileResponseDTO getProfile();

    AdminProfileResponseDTO updateProfile(
            UpdateProfileRequestDTO request);

}